File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    # 캐시구성 => 오래된것 --- 새로운 것 순 
    Cache = []
    Time = 0
    for city in cities:
        city = city.lower() # 공백 특수 없고, 대소문자가 없으니
        if city in Cache:
            Cache.remove(city)
            Cache.append(city)
            Time += 1
        else: #miss
            Cache.append(city)
            if cacheSize < len(Cache):
                Cache.pop(0)
            Time += 5
    return Time

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;

public class Solution {
    public int solution(int cacheSize, String[] cities) {
        if (cacheSize == 0) return cities.length * 5;

        int time = 0;
        LinkedHashMap<String, Boolean> cache = new LinkedHashMap<>(cacheSize, 0.75f, true);

        for (String city : cities) {
            city = city.toLowerCase();

            if (cache.containsKey(city)) {
                time += 1;
                cache.get(city);
            } else {
                time += 5;
                if (cache.size() >= cacheSize) {
                    String lru = cache.keySet().iterator().next();
                    cache.remove(lru);
                }
                cache.put(city, true);
            }
        }

        return time;
    }
}